server/plugin/api_test.go

```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
+			},
+			ExpectedStatusCode: http.StatusBadRequest,
+		},
+		"invalid record type": {
+			RequestBody: `{
+		 		"sys_id": "mockSysID",
+		 		"record_type": "testRecordType"
+		 		}`,
+			ChannelID: "bnqnzipmnir4zkkj95ggba5pde",
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("LogError", mock.AnythingOfType("string"), "Record type", "testRecordType").Return()
+				return api
+			},
+			ExpectedStatusCode:   http.StatusBadRequest,
+			ExpectedErrorMessage: constants.ErrorInvalidRecordType,
+		},
+		"failed to get the user": {
+			RequestBody: `{
+				"sys_id": "mockSysID",
+				"record_type": "incident"
+				}`,
+			ChannelID: "bnqnzipmnir4zkkj95ggba5pde",
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("GetUser", mock.AnythingOfType("string")).Return(
+					nil, testutils.GetInternalServerAppError(),
+				)
+
+				api.On("LogError", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+				return api
+			},
+			ExpectedStatusCode: http.StatusInternalServerError,
+		},
+		"failed to create the post": {
+			RequestBody: `{
+				"sys_id": "mockSysID",
+				"record_type": "incident"
+				}`,
+			ChannelID: "bnqnzipmnir4zkkj95ggba5pde",
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("GetUser", mock.AnythingOfType("string")).Return(
+					testutils.GetUser(), nil,
+				)
+
+				api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(
+					nil, testutils.GetInternalServerAppError(),
+				)
+
+				api.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...).Return()
+				return api
+			},
+			ExpectedStatusCode: http.StatusOK,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			api := test.SetupAPI(&plugintest.API{})
+			defer api.AssertExpectations(t)
+			defer monkey.UnpatchAll()
+
+			p := setupTestPlugin(api, nil)
+			_ = setupPluginForCheckOAuthMiddleware(p, t)
+
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest(http.MethodPost, strings.Replace(requestURL, "{channel_id:[A-Za-z0-9]+}", test.ChannelID, 1), bytes.NewBufferString(test.RequestBody))
+			r.Header.Add(constants.HeaderMattermostUserID, testutils.GetID())
+			p.ServeHTTP(nil, w, r)
+
+			result := w.Result()
+			require.NotNil(t, result)
+			defer result.Body.Close()
+
+			assert.Equal(test.ExpectedStatusCode, result.StatusCode)
+			if test.ExpectedErrorMessage != "" {
+				var resp *serializer.APIErrorResponse
+				err := json.NewDecoder(result.Body).Decode(&resp)
+				require.Nil(t, err)
+
+				assert.Contains(resp.Message, test.ExpectedErrorMessage)
+			}
+		})
+	}
+}
+
+func TestGetCommentsForRecord(t *testing.T) {
+	requestURL := fmt.Sprintf("%s%s", constants.PathPrefix, constants.PathCommentsForRecord)
+	requestURL = strings.Replace(requestURL, "{record_id:[0-9a-f]{32}}", testutils.GetServiceNowSysID(), 1)
+	for name, test := range map[string]struct {
+		RecordType           string
+		SetupAPI             func(*plugintest.API) *plugintest.API
```

Comment:
```suggestion
		SetupAPI             func(*plugintest.API)
```
Can you make this change in the whole file? It should work just like the SetupClient function is working.